frontend/src/contexts/appContext.tsx

```diff
@@ -0,0 +1,43 @@
+import { createContext, useState, useCallback, useRef, useEffect } from 'react';
+import type { PropsWithChildren } from 'react';
+import { ActivityIndicator, Modal, SafeAreaView, Animated } from 'react-native';
+
+type appContext = { startLoading: () => void, stopLoading: () => void };
+const AppContext = createContext<appContext | null>(null);
+
+export const AppContextProvider = ({ children }: PropsWithChildren): JSX.Element => {
+    const [isLoading, setIsLoading] = useState(false);
+    const startLoading = useCallback(() => {
+        setIsLoading(true);
+    }, [setIsLoading])
+    const stopLoading = useCallback(() => {
+        setIsLoading(false);
+    }, [setIsLoading])
+
+    const opacityAnim = useRef(new Animated.Value(1)).current;
+    const loadingOpacity = 0.8 // should brainstorm and decide on a value
+    useEffect(() => {
```

Comment:
is there a way to potentially do this without useEffect? I am just worried about performance since it might be being used everywhere and may cause unnessary rerenders in a lot of different places. I could be wrong but

Reply:
Yes we could get rid of it then put it into the start and stop loading functions, but I doubt we should be worried about anything performance wise for this since only the Animated view will re-render regardless of useEffect or not.

Reply:
Actually I think I found an even better alternative to using Animated View, will show you later in our meeting